### ISSUE TITLE
Instance: Make firewall rules more visible 

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/console#2064]
Users new to us are not necessarily aware of the need to add/modify firewall rules in order to access their applications outside of the rack. The "VPC" name on the instance page (or Create Instance page) also does not suggest anything related to security for that matter. Here are two suggestions for users to have a better chance of getting the firewall rules right:

1. Create Instance: provide a link on/near the VPC name that lands on the Firewall Rules tab selected by the user (perhaps open a new tab so user can stay on Create Instance); this allows user to confirm if the VPC selected is right for them.
2. Instance details: provide a similar link to the VPC Firewall Rules or add another subtab (named more generally as "Security", similar to other public cloud portals) that actually displays all the rules in that VPC.

**Comment by david-crespo** [comment]
Ben did some related experimentation in #1890

**Comment by askfongjojo** [comment]
Thanks for pointing me to the PR! (I only searched for related issues but totally forgot about open PRs before I filed this.) It covers exactly the second item in this ticket and great discussions about it. I second the idea of _not_ making edit rules available on the instance tab (to prevent user from inadvertently changing the rules on other instances) and _not_ trying to determine which rules apply to the instance (for the various reasons stated in the PR comments).

**Comment by charliepark** [comment]
One possibility: VPC link on instance detail table says "VPC / Firewall Rules"

**Comment by charliepark** [comment]
Here's what that looks like …
<img width="1227" alt="Screenshot 2024-05-09 at 1 13 28 PM" src="https://github.com/oxidecomputer/console/assets/22547/8bad2a78-5809-4a67-9771-98dba8abb6b9">

**Comment by charliepark** [comment]
This does not _need_ to be in Release 9, and we don't have a compelling solution to the problem, so I removed the `9` Milestone tag.

**Comment by david-crespo** [comment]
Relevant experience report: https://github.com/oxidecomputer/docs/issues/801

I'm thinking we could put a link to the default VPC's firewall rules in the networking tab on the instance. The idea from above about mentioning firewall rules in the VPC label would also work better now that the label is not in a badge:

<img width="411" height="198" alt="Image" src="https://github.com/user-attachments/assets/fb7659b7-cc10-43a5-87f0-2da69a98049f" />

**Comment by charliepark** [comment]
One possibility: <img width="1131" height="784" alt="Image" src="https://github.com/user-attachments/assets/94ed1e78-df04-4914-9dae-1ba20c14d541" />

**Comment by david-crespo** [comment]
Yeah, I think that’s pretty good. I almost wish we had a message box color that was even more neutral than that, but it’s ok. I’d probably shorten the copy. I was also thinking we could add a box like the external IPs and network interfaces ones but inside it, instead of a table we have that message. Kind of janky but fits with the page and lets us justify putting it at the bottom rather than the top. Or maybe it doesn’t have to use the blue message, it could just be some normal text in place of the table. “Edit firewall rules on the VPC”. One worry I have is that there are multiple VPCs — maybe if there’s one we link directly to it but if there are more we link to the list?

**Comment by charliepark** [comment]
If multiple VPCs, we can tweak the link copy a bit (in addition to sending them to the VPC list as opposed to the firewall rules tab).
<img width="1204" height="1123" alt="Image" src="https://github.com/user-attachments/assets/83b3c6e3-0191-4d9f-849c-7ed269c2d52a" />
<img width="1199" height="1181" alt="Image" src="https://github.com/user-attachments/assets/6581f776-2756-4115-80a6-69f38e40f401" />

I also tried a more subtle treatment, but I think it gets lost:
<img width="1209" height="1151" alt="Image" src="https://github.com/user-attachments/assets/5461d1a9-84c1-4b50-bec1-ff52196b2a3b" />

**Comment by rcgoodfellow** [comment]
I think having a link to the VPC firewall rules from the instance networking page is a good start and will provide immediate benefit to users. An instance will only ever belong to one VPC, so we can start by linking to that VPC. 

When we get to the VPC routing feature, I think that there are a number of things we'll want to think about in terms of providing a holistic view of connectivity, and firewall rules would be a part of that.

**Comment by benjaminleonard** [comment]
> An instance will only ever belong to one VPC, so we can start by linking to that VPC.

Would this be the VPC that is attached to an instances primary NIC?

**Comment by rcgoodfellow** [comment]
Yeah. All interfaces/NICs on an instance currently are required to be in the same VPC.
